### PR TITLE
Add CIS 1.14

### PR DESCRIPTION
**Pull request by orouz**
### Summary of your changes

Adds CIS GCP 1.14

| failed (restrictions on `cloudapis.googleapis.com`) |  passed (no restrictions on `cloudapis.googleapis.com`)  |
|-----------|---|
| AFAIK editing api keys doesn't allow adding this restriction |  Screenshot 2023-08-01 at 15 31 04|

 

### Related Issues

- https://github.com/elastic/security-team/issues/6179

### Checklist

- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary README/documentation (if appropriate)

#### Introducing a new rule?
- [x] Generate rule metadata using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rules-metadata)
- [x] Add relevant unit tests
- [ ] Generate relevant rule templates using [this script](https://github.com/elastic/csp-security-policies/tree/main/dev#generate-rule-templates), and open a PR in [elastic/packages/cloud_security_posture](https://github.com/elastic/integrations/tree/main/packages/cloud_security_posture)

> **Note** Make sure to bump cloudbeat [`version/policy.go`](https://github.com/elastic/cloudbeat/blob/main/version/policy.go#L20) after merging this PR and drafting a new release.

### CODE REVIEW

File: bundle/compliance/cis_gcp/rules/cis_1_14/rule.rego [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+import data.compliance.policy.gcp.data_adapter
+import future.keywords.if
+
+default is_valid = false
+
+restrictions if {
+	data_adapter.resource.data.restrictions
+}
+
+is_valid if {
```
Comment (uri-weisman) [inline bundle/compliance/cis_gcp/rules/cis_1_14/rule.rego:13]: what happens if the `restrictions` or `apiTargets` fields do not exist?
Is this a valid scenario?

Comment (uri-weisman) [reply bundle/compliance/cis_gcp/rules/cis_1_14/rule.rego:13]: > Each key should have a line that says restrictions: followed by varying parameters and NOT have a line saying - service: cloudapis.googleapis.com

If I understand correctly the rule, we need to verify that at least a single restriction exists and verify that it is not the `- service: cloudapis.googleapis.com`

**Comment by uri-weisman** [comment]
@orouz - you can create a permissive key using the following command.
```
gcloud alpha services api-keys create --display-name=test-permissive-key --api-target=service=cloudapis.googleapis.com
```

https://console.cloud.google.com/apis/credentials/key/7c00d1c4-0d13-408d-98ed-2ab92e05244c?project=elastic-security-test&supportedpurview=project

File: bundle/compliance/cis_gcp/rules/cis_1_14/rule.rego
Comment (oren-zohar) [inline bundle/compliance/cis_gcp/rules/cis_1_14/rule.rego:1]: can you replace the `is_valid` pattern here too?

**Review by oren-zohar — CHANGES_REQUESTED** [review]

**Review by oren-zohar — APPROVED** [review]

File: bundle/compliance/cis_gcp/rules/cis_1_14/test.rego [hunk 2]
```diff
@@ -9,10 +9,16 @@ type := "key-management"
...
 	eval_fail with input as test_data.generate_gcp_asset(
-		type, subtype, {"data": {"restrictions": {"apiTargets": [{"service": "cloudapis.googleapis.com"}]}}},
+		type, subtype, {"data": {}},
 		{},
 	)
+	# # fails when google apis are restricted
+	# eval_fail with input as test_data.generate_gcp_asset(
+	# 	type, subtype, {"data": {"restrictions": {"apiTargets": [{"service": "cloudapis.googleapis.com"}]}}},
+	# 	{},
+	# )
```
Comment (uri-weisman) [inline bundle/compliance/cis_gcp/rules/cis_1_14/test.rego:21]: @orouz should be commented out?

Comment (orouz) [reply bundle/compliance/cis_gcp/rules/cis_1_14/test.rego]: yeah missed it, fixed now. thanks!

**Pull request merged**